scripts/src/javatests/org/oppia/android/scripts/xml/TextViewStyleCheckTest.kt

```diff
@@ -0,0 +1,331 @@
+package org.oppia.android.scripts.xml
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+
+/** Tests for [TextViewStyleCheck]. */
+class TextViewStyleCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+  private val TEXTVIEW_STYLE_CHECK_PASSED_OUTPUT_INDICATOR = "TEXTVIEW STYLE CHECK PASSED"
+  private val TEXTVIEW_STYLE_CHECK_FAILED_OUTPUT_INDICATOR = "TEXTVIEW STYLE CHECK FAILED"
+
+  @field:[Rule JvmField] val tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("app", "src", "main", "res")
+    tempFolder.newFolder("app/src/main/res/layout")
+    tempFolder.newFolder("app/src/main/res/values")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testTextViewStyle_validStyleAttribute_checksPass() {
+    val validLayout =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+        <TextView
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"
+          style="@style/ValidTextViewStyle"/>
+      </LinearLayout>
+      """.trimIndent()
+
+    createLayoutFile(validLayout)
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(TEXTVIEW_STYLE_CHECK_PASSED_OUTPUT_INDICATOR)
+  }
+
+  @Test
+  fun testTextViewStyle_missingStyleAttribute_checksFail() {
+    val invalidLayout =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+        <TextView
+          android:id="@+id/test_text_view_no_style"
+          android:layout_width="wrap_content"
+          android:layout_height="wrap_content"/>
+      </LinearLayout>
+      """.trimIndent()
+
+    createLayoutFile(invalidLayout)
+
+    val thrown = assertThrows<Exception>() { runScript() }
```

Comment:
Nit, here and below, but I feel like good variable names are important.
```suggestion
    val exception = assertThrows<Exception>() { runScript() }
```